### ISSUE TITLE
TIL: don't apply weight decay (L2/L1/any regulizer) to a batchnorm layers (Modules ) and also bias.

### ISSUE THREAD

**Issue by carlesoctav** [carlesoctav/ml-stuffs#1]
To put simply, there are weights and bias in a batchnorm formula:
$$
(\frac{(x - \mu)}{\sigma}) * a + b
$$
on an initialization the value of a will be a Tensor(1) and b (bias) will be a Tensor(0).
If you're using a weight-decay on this parameters, it will actually force a, b to a zero value.

full discussion:
https://discuss.pytorch.org/t/weight-decay-in-the-optimizers-is-a-bad-idea-especially-with-batchnorm/16994/3
